src/CMakeLists.txt

```diff
@@ -1,3 +1,3 @@
 add_subdirectory(esp32)
```

Comment:
Not sure why we keep this library, shoudn't the code here be in the posix bsp instead?

Reply:
I also find it a little awkward having POSIX code inside of a esp32 folder

Reply:
This is the equivalent of the interface linker library mentioned in the comment on src/main.cpp.  I will move it to the posix bsp for clarity.

Reply:
Moved